src/Mantasflowers.Services/Services/Order/OrderService.cs

```diff
@@ -12,24 +14,34 @@ namespace Mantasflowers.Services.Services.Order
     public class OrderService : IOrderService
     {
         private readonly IUnitOfWork _unitOfWork;
+        private readonly IHashMapService _hashMapService;
         private readonly IMapper _mapper;
 
-        public OrderService(IUnitOfWork unitOfWork, IMapper mapper)
+        public OrderService(
+            IUnitOfWork unitOfWork,
+            IHashMapService hashMapService,
+            IMapper mapper)
         {
             _unitOfWork = unitOfWork;
+            _hashMapService = hashMapService;
             _mapper = mapper;
         }
 
         public async Task<Domain.Entities.Order> CreateOrderAsync(PostCreateOrderRequest request)
         {
             var order = _mapper.Map<Domain.Entities.Order>(request);
-            order.TemporaryPasswordHash = "stoppolice";
+
+            (string uniquePassword, string passwordHash) = PasswordGenerator.HashUniquePassword();
+
+            order.UniquePassword = uniquePassword;
 
             try
             {
                 order = await _unitOfWork.OrderRepository.CreateAsync(order);
+                var hashMap = await _hashMapService.CreateHashMapAsync(passwordHash);
+                hashMap.Order = order;
             }
-            catch (DbUpdateException)
+            catch (ArgumentNullException)
```

Comment:
I would just explicitly check for null where needed (guessing `hashMap.Order`) instead of catching such a general exception.

Reply:
There is a reason. 
```cs
public async override Task<Order> CreateAsync(Order entity)
{
    foreach (var item in entity.OrderItems)
    {
        item.Product = await _dbContext.Products.FindAsync(item.ProductId);
        _dbContext.Entry(item.Product).State = EntityState.Unchanged;
    }

    entity = await base.CreateAsync(entity);

    return entity;
}
```
Given product ids, it will throw if any of the products do not exist (which is a very likely case). Imagine a case, where product gets deleted, but front end sends its id.